src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -1,19 +1,32 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
 import {
     SaveAssessmentButton,
     SaveAssessmentButtonProps,
 } from 'DetailsView/components/save-assessment-button';
 import { shallow } from 'enzyme';
 import * as React from 'react';
+import { IMock, Mock, MockBehavior } from 'typemoq';
 
 describe('SaveAssessmentButton', () => {
-    let props: SaveAssessmentButtonProps;
+    let propsStub: SaveAssessmentButtonProps;
+    let detailsViewActionMessageCreatorMock: IMock<DetailsViewActionMessageCreator>;
 
-    it('should render per the snapshot', () => {
-        const rendered = shallow(
-            <SaveAssessmentButton {...props} download={'download'} href={'url'} />,
+    beforeEach(() => {
+        detailsViewActionMessageCreatorMock = Mock.ofType(
+            DetailsViewActionMessageCreator,
+            MockBehavior.Strict,
         );
+        propsStub = {
+            deps: { detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object },
+            download: 'download',
+            href: 'url',
+        };
+    });
+
+    it('should render per the snapshot', () => {
+        const rendered = shallow(<SaveAssessmentButton {...propsStub} />);
```

Comment:
We should also validate that onClick is being passed correct function by simulating a click and verifying using the mock.